connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerTest.java

```diff
@@ -3025,24 +3032,26 @@ private void verifyVersionedTaskHeaderConverterFromWorker() {
         verify(plugins).newHeaderConverter(any(WorkerConfig.class), eq(WorkerConfig.HEADER_CONVERTER_CLASS_CONFIG), eq(WorkerConfig.HEADER_CONVERTER_VERSION));
     }
 
-    private void mockGenericIsolation() {
-        when(plugins.withClassLoader(pluginLoader)).thenReturn(loaderSwap);
+    private void mockGenericIsolation(boolean connector) {
+        if (!connector) {
+            when(plugins.withClassLoader(null)).thenReturn(loaderSwap);
```

Comment:
this doesn't look right; If we're calling withClassLoader with a null value, it means that some other call is returning a null ClassLoader, which is suspicious of a bug or incomplete mocking.

Reply:
You're right, fixed